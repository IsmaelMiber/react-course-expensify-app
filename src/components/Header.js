import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ul>
            <li> <NavLink to="/" activeClassName="is-active" exact={true} >Home</NavLink> </li>
            <li> <NavLink to="/create" activeClassName="is-active" exact={true} >Create Expense Page</NavLink> </li>
            <li> <NavLink to="/help" activeClassName="is-active" exact={true} >Help Page</NavLink> </li>
        </ul>
    </header>
);

export default Header;