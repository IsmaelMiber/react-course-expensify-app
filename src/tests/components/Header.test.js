// Shallow Rendering
    // 1. Not Care for Children Components.
    // 2. Note Care for User Interaction and Life-Cycles Events.
// Full DOM Rendering
    // 1. Care for Children Components. (render them also)
    // 2. Care for User Interaction and Life-Cycles Events.

import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test("Should Header Component render correctly", () => {
   const wrapper = shallow(<Header />);
   expect(wrapper).toMatchSnapshot();
});