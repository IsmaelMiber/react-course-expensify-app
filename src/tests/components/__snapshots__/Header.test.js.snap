// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header Component render correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ul>
    <li>
       
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
       
    </li>
    <li>
       
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/create"
      >
        Create Expense Page
      </NavLink>
       
    </li>
    <li>
       
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/help"
      >
        Help Page
      </NavLink>
       
    </li>
  </ul>
</header>
`;
